src/js/controller.js

```diff
@@ -73,17 +65,23 @@ export default class Controller extends Component {
                 }
             }
 
-            if (!isEqual(hiddenIndicators, this._hiddenIndicators)) {
-                this._hiddenIndicators = hiddenIndicators;
-                this.reloadDataMapper();
-                this.triggerEvent('my_view.hiddenIndicatorsPanel.reload', this.hiddenIndicators);
-            }
-
-            if (!isEqual(filteredIndicators, this._filteredIndicators)) {
-                this._filteredIndicators = filteredIndicators;
-                this.triggerEvent('my_view.filteredIndicators.updated', this.filteredIndicators);
-                this.triggerEvent(VersionController.EVENTS.ready, this.versionController.allVersionsBundle);
-                this.reDrawChildren();
+            if (areaCode !== this.versionController.areaCode){
+              let payload = this.changeGeography(areaCode)
+              this._shouldMapZoom = true;
+              this.onHashChange(payload);
```

Comment:
Why do we call onHashChange here? isn't it a bit weird to go through hashchange when the point is to move away from hashchange for loading the current geo?
If it does make sense still, perhaps a comment explaining why might be helpful a year from now.